phpseclib/Crypt/RSA.php

```diff
@@ -2053,17 +2033,20 @@ function _rsassa_pkcs1_v1_5_verify($m, $s)
         $s = $this->_os2ip($s);
         $m2 = $this->_rsavp1($s);
         if ($m2 === false) {
-            throw new \RuntimeException('Invalid signature');
+            return false;
         }
         $em = $this->_i2osp($m2, $this->k);
         if ($em === false) {
-            throw new \RuntimeException('Invalid signature');
+            return false;
         }
 
         // EMSA-PKCS1-v1_5 encoding
 
-        $em2 = $this->_emsa_pkcs1_v1_5_encode($m, $this->k);
-        if ($em2 === false) {
+        // If the encoding operation outputs "intended encoded message length too short," output "RSA modulus
+        // too short" and stop.
+        try {
+            $em2 = $this->_emsa_pkcs1_v1_5_encode($m, $this->k);
+        } catch (\LengthException $e) {
```

Comment:
This does not account for `_emsa_pkcs1_v1_5_encode()` possibly returning `false` (in https://github.com/phpseclib/phpseclib/blob/806249c7e91b60e81ae3052cf02a83b57ac74c7d/phpseclib/Crypt/RSA.php#L1944). Is that intentional?